net/batman-adv-legacy/Makefile

```diff
@@ -98,7 +97,7 @@ $(eval $(call Download,batctl-legacy))
 BATCTL_EXTRACT = tar xzf "$(DL_DIR)/batctl-$(BATCTL_VERSION).tar.gz" -C "$(PKG_BUILD_DIR)"
 BATCTL_PATCH = $(call Build/DoPatch,"$(PKG_BATCTL_BUILD_DIR)","$(PATCH_DIR)",batctl)
 BATCTL_BUILD = $(MAKE_BATCTL_ENV) $(MAKE) -C $(PKG_BATCTL_BUILD_DIR) $(MAKE_BATCTL_ARGS)
-BATCTL_INSTALL = $(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/local/sbin/batctl $(1)/usr/sbin/
+BATCTL_INSTALL = $(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/local/sbin/batctl $(1)/usr/sbin/batctl-legacy
```

Comment:
Hmm, I've been wondering: How well does batadv-legacy work with the current master batctl? We have backported some parts of the netlink API, so possibly we can get rid of the old batctl for good and only use the regular batctl package?

Reply:
Good question. Yes, the current batctl seems to work quite well.

What made me worry a bit were the differences in header numbers for various outputs. So compatibility regarding "batctl <cmd> -H" - which is used by gluon-ebtables-limit-arp. It seems however, that we are lucky:

Using batman-adv-legacy.ko:

<pre><code>root@nsm2-loco-test:~# /usr/sbin/batctl dc
Distributed ARP Table (bat0):
          IPv4             MAC           last-seen

root@nsm2-loco-test:~# /usr/sbin/batctl-legacy  dc
Distributed ARP Table (bat0):
          IPv4             MAC           last-seen
</code></pre>

And:

<pre><code>root@nsm2-loco-test:~# /usr/sbin/batctl-legacy tl
Locally retrieved addresses (from bat0) announced via TT (TTVN: 1 CRC: 0x5e28):
       Client        Flags   Last seen
 * 6e:c7:a4:0a:c8:45 [.P...]   0.000

root@nsm2-loco-test:~# /usr/sbin/batctl tl
[B.A.T.M.A.N. adv 2013.4.0, MainIF/MAC: dummy0/b6:27:2c:16:14:25 (bat0/6e:c7:a4:0a:c8:45 BATMAN_IV), TTVN: 1]
Client             VID Flags    Last seen (CRC       )
6e:c7:a4:0a:c8:45   -1 [.P....]   0.000   (0x00005e28)

root@nsm2-loco-test:~# /usr/sbin/batctl-legacy tl -H
 * 6e:c7:a4:0a:c8:45 [.P...]   0.000

root@nsm2-loco-test:~# /usr/sbin/batctl tl -H
6e:c7:a4:0a:c8:45   -1 [.P....]   0.000   (0x00005e28)
</code></pre>

So at least for "tl" and "dc" which are used by gluon-ebtables-limit-arp all have two header lines.

Reply:
And actually, for netlink the fixed header line numbers aren't used, right. So even if the header lines would differ between batman-adv and batman-adv-legacy, that would not matter for us (except for the dat_cache table - seems we haven't backported the dat-cache netlink support to batman-adv-legacy yet).